fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/OcflPersistentStorageUtils.java

```diff
@@ -180,9 +181,9 @@ public static String resolveExtensions(final String subPath, final boolean isRdf
      * @param fedoraIdentifier The fedora identifier
      * @return The resolved topic
      */
-    public static  String resolveTopic(final String fedoraIdentifier) {
-        if (fedoraIdentifier.endsWith(FEDORA_METADATA_SUFFIX)) {
-            return fedoraIdentifier.substring(0, fedoraIdentifier.indexOf(FEDORA_METADATA_SUFFIX));
+    public static FedoraId resolveTopic(final FedoraId fedoraIdentifier) {
```

Comment:
It would seem that this method (which can be `private`) could be replaced by `FedoraId.asBaseId()`?

Reply:
> 
> 
> Looks good. Some minor comments.
> As a note, it is becoming slightly confusing when to use which method of `FedoraId`. Can we work towards removing some of the "getters" that return Strings?

I don't disagree! Because the id is used differently in different contexts it is confusing to know which method you want or even what the current "state" of the id is. I attempted to increase the clarity in this recent PR: https://github.com/fcrepo4/fcrepo4/pull/1719, but it's still not perfect.